packages/next/src/lib/format-server-error.ts

```diff
@@ -21,6 +21,20 @@ function setMessage(error: Error, message: string): void {
 }
 
 export function formatServerError(error: Error): void {
+  if (
+    error.message.includes(
+      'Class extends value undefined is not a constructor or null'
+    )
+  ) {
+    setMessage(
+      error,
+      `${error.message}
+
+This might be caused by a React Class component being rendered in a server component, React Class components only works in client components. Read more: https://nextjs.org/docs/messages/class-component-in-server-component`
```

Comment:
```suggestion
This might be caused by a React Class component being rendered in a server component, React Class components only works in Client Components. Read more: https://nextjs.org/docs/messages/class-component-in-server-component`
```